### PR TITLE
Week8

### PR DESCRIPTION
**Pull request by ManuelaFlores**
Hi Fuad, 
Here are my changes for week8.

Regards

### CODE REVIEW

File: week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            .setRequiredNetworkType(NetworkType.NOT_ROAMING)
+            .build()
+
+        val synchronizeServiceWorker =
+            PeriodicWorkRequestBuilder<SynchronizeServiceWorker>(1, TimeUnit.HOURS)
+                .setConstraints(constraint)
+                .build()
+
+        val workManager = WorkManager.getInstance(this)
+        workManager.enqueueUniquePeriodicWork(
```
Comment (filbabic) [inline week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/ui/main/MainActivity.kt:39]: Nicely done here!

File: week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/worker/SynchronizeServiceWorker.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+const val NOTIFICATION_CHANNEL_ID = "Synchronize ID"
+
+class SynchronizeServiceWorker(context: Context, workerParameters: WorkerParameters) :
+    CoroutineWorker(context, workerParameters) {
+
+    private val apiService by lazy { buildApiService() }
+
+    private val remoteApi by lazy { RemoteApiManager(apiService) }
+
+    private val photosDao: PhotosDao =
```
Comment (filbabic) [inline week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/worker/SynchronizeServiceWorker.kt:26]: You could've exposed some of these functions to build the dao/repo in the App class, and then just call a single function!

**Review by filbabic — APPROVED** [review]
Only two small comments! I'd love for you to take some of the feedback from the Week 7 assignment and then fix this for the next few weeks, as these are things that'll help you work on some of the future assignments.

**Good Job** on the assignment!

File: week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/worker/SynchronizeServiceWorker.kt (hunk 2 above)
Comment (ManuelaFlores) [reply week8/PhotoInspiration/app/src/main/java/com/manuflowers/photoinspiration/worker/SynchronizeServiceWorker.kt:26]: Done

**Pull request merged**